server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/FlowInstanceController.java

```diff
@@ -176,6 +176,15 @@ public SuccessResponse<String> getLog(@PathVariable Long id, @RequestParam OdcTa
         return Responses.single(flowTaskInstanceService.getLog(id, logType));
     }
 
+    @ApiOperation(value = "downloadLog", notes = "下载任务完整日志")
+    @RequestMapping(value = "/{id:[\\d]+}/tasks/log/download", method = RequestMethod.GET)
+    public ResponseEntity<InputStreamResource> downloadLog(@PathVariable Long id) throws IOException {
```

Comment:
It may cause an OutOfMemoryError when the log file is too large, as the file is read into the JVM and then sent to the front-end in this manner.

Reply:
We use `InputStreamResource ` and the inner inplementation is `FileInputStream`, so it will not occupied too much memory. I have test it by download three 1GB log files and the memory rate is only 2.8%:
<img width="803" alt="image" src="https://github.com/oceanbase/odc/assets/84772536/7ea736bd-af98-4787-ba4d-1f385027c502">